Animals/src/test/java/AnimalRepositoryImplTest.java

```diff
@@ -0,0 +1,97 @@
+import Model.AnimalClasses.Pets.Cat;
+import Model.AnimalClasses.Pets.Dog;
+import Model.AnimalClasses.Predators.Shark;
+import Model.AnimalClasses.Predators.Wolf;
+import Repository.AnimalRepositoryImpl;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class AnimalRepositoryImplTest {
+    List<Animal> animalList;
+    AnimalRepositoryImpl animalRepository = new AnimalRepositoryImpl();
+
+    @DisplayName("Кейс, когда хотя бы у одного из животных год високостный")
+    @Test
+    void findLeapYearNamesPositive() throws InvalidAnimalBirthDateException {
+        animalList = new ArrayList<>();
+        Dog dog = new Dog("Немецкая овчарка", "Елисей", 999.99, "Не кусает за бочок", LocalDate.of(2016, 10, 6));
+        Wolf wolf = new Wolf("Серый Волк", "Мухтар", 888.88, "Кусает за бочок", LocalDate.of(2021, 3, 17));
+        animalList.add(dog);
+        animalList.add(wolf);
+        assertFalse(animalRepository.findLeapYearNames(animalList).isEmpty());
+    }
+
+    @DisplayName("Кейс, когда никого из животных год високостный")
+    @Test
+    void findLeapYearNamesNegative() throws InvalidAnimalBirthDateException {
+        animalList = new ArrayList<>();
+        Dog dog = new Dog("Немецкая овчарка", "Елисей", 999.99, "Не кусает за бочок", LocalDate.of(2017, 10, 6));
+        Wolf wolf = new Wolf("Серый Волк", "Мухтар", 888.88, "Кусает за бочок", LocalDate.of(2021, 3, 17));
+        animalList.add(dog);
+        animalList.add(wolf);
+        assertTrue(animalRepository.findLeapYearNames(animalList).isEmpty());
+    }
+
+    @DisplayName("Кейс, когда найдены животные старше указанного года")
+    @Test
+    void findOlderAnimalPositive() {
+        animalList = new ArrayList<>();
+        Dog dog = new Dog("Немецкая овчарка", "Елисей", 999.99, "Не кусает за бочок", LocalDate.of(2016, 10, 6));
+        Wolf grayWolf = new Wolf("Серый Волк", "Мухтар", 888.88, "Кусает за бочок", LocalDate.of(2021, 3, 17));
+        Wolf whiteWolf = new Wolf("Белый Волк", "Шарик", 777.77, "Замыкает стаю", LocalDate.of(2023, 3, 17));
+        animalList.add(dog);
+        animalList.add(grayWolf);
+        animalList.add(whiteWolf);
+        assertEquals(animalRepository.findOlderAnimal(animalList, 2).size(), 2);
+    }
+
+    @DisplayName("Кейс, когда найдены животные старше указанного года, выводится самый старший")
+    @Test
+    void findOlderAnimalNegative() {
+        animalList = new ArrayList<>();
+        Dog dog = new Dog("Немецкая овчарка", "Елисей", 999.99, "Не кусает за бочок", LocalDate.of(2017, 10, 6));
+        Wolf wolf = new Wolf("Серый Волк", "Мухтар", 888.88, "Кусает за бочок", LocalDate.of(2021, 3, 17));
+        animalList.add(dog);
+        animalList.add(wolf);
+        assertEquals(animalRepository.findOlderAnimal(animalList, 10).size(), 1);
```

Comment:
Проверь не просто размер списк, а что именно старший содержиться в этом списке

Reply:
Вот так норм вроде. Тест пройден успешно
```suggestion
        assertEquals(Period.between(dog.getBirthDate(), LocalDate.now()).getYears(), animalRepository.findOlderAnimal(animalList, 10).get(dog));
```